include/mqt-core/fomac/FoMaC.hpp

```diff
@@ -674,12 +709,13 @@ class FoMaC {
 
 private:
   QDMI_Session session_ = nullptr;
+  mutable std::mutex mutex_;
+  bool initialized_ = false;
+  std::unordered_map<SessionParameter, std::string> pendingParameters_;
```

Comment:
In order to stay consistent with how we handled other aspects of QDMI in the C++ interface, I would argue that the session parameters should either be individual member functions or they should be keyword arguments in the constructor of a session. I would have a slight preference for the kwarg approach as it is very close to what we would expect to expose to Python/Qiskit.

Reply:
The question to me is: from a practical standpoint, would there ever be a case where a user might sequentially set authentication information because it is not known in its entirety apriori?
But now that I think of it, the kwargs solution is only convenient in Python, where the order of the function parameters doesn't matter as long as you use named arguments.
For C++, a function that takes 8 parameters which are all optional is quite cumbersome to use if you, e.g., want to only set the 8th parameter.

How bad would it be to expose a different API for C++ and Python? 🙃

Reply:
I like that approach 👍🏻